src/transform/tests/testdata/flatmap-to-map

```diff
@@ -0,0 +1,43 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Test the FlatMapToMap transform
+
+cat
+(defsource x [int64])
+----
+ok
+
+# Rewrite is possible in this case
+build apply=FlatMapToMap
+(FlatMap
+    (get x)
+    (Wrap [(String false)] 1) [("a_string" String)])
+----
+%0 =
+| Get x (u0)
+| Map "a_string"
+
+# Rewrite possible
+build apply=FlatMapToMap
+(FlatMap
+    (get x)
+    (Wrap [(String false)] 3) [("a_string" String) ("another" String) ("test" String)])
+----
+%0 =
+| Get x (u0)
+| Map "a_string", "another", "test"
+
+# Produces more than one row, must not rewrite this
+build apply=FlatMapToMap
+(FlatMap (get x) (Wrap [(Int32 false)] 1) [(0 Int32) (1 Int32) (2 Int32)])
```

Comment:
Can we have a test where the `Wrap` type is not a singleton - maybe a `[(Int32 false) (String false)]`?